Azure.Data.Wrappers/StorageQueue.cs

```diff
@@ -155,6 +156,23 @@ public virtual CloudQueue Reference
             return await this.reference.GetMessagesAsync(messageCount, this.visibilityTimeout, null, null);
         }
 
+        public virtual async Task<T> GetAsync<T>()
+        {
+            var returned = await this.reference.GetMessageAsync(this.visibilityTimeout, null, null);
+            return JsonConvert.DeserializeObject<T>(returned.AsString);
+        }
+
+        public async Task<IEnumerable<T>> GetManyAsync<T>(int messageCount = 5)
+        {
+            if (0 >= messageCount)
+            {
+                messageCount = 1;
+            }
```

Comment:
Why handle this instead of throwing or letting the called method fail?  This seems like an assumption of use.

Reply:
Fair, but I didn't want to change current behavior